update-netty-substitutions.sh

```diff
@@ -1,33 +1,53 @@
 #!/bin/bash
 
-# Set directories as variables
+###################################################################################
+# Script: update-netty-substitutions.sh
+# Purpose: Updates and manages Netty and gRPC substitutions for Temporal extension
+#          by copying and modifying relevant files from Quarkus codebase.
+#
+# This script performs the following main tasks:
+# 1. Clones specific version of Quarkus repository
+# 2. Copies Netty and gRPC related files to appropriate directories
+# 3. Updates namespace references for shaded Netty
+# 4. Fixes imports and package declarations
+# 5. Updates configuration handling for compatibility
+###################################################################################
+
+# Directory structure for source files in Quarkus repository
+# These paths are relative to the Quarkus root directory
 SRC_NETTY_RUNTIME="quarkus/extensions/netty/runtime/src/main/java/io/quarkus/netty"
 SRC_GRPC_RUNTIME="quarkus/extensions/grpc-common/runtime/src/main/java/io/quarkus/grpc/common"
 
-DEST_NETTY_RUNTIME="extension/runtime/src/main/java/io/quarkiverse/temporal/graal/nettyhandling"
-DEST_GRPC_RUNTIME="extension/runtime/src/main/java/io/quarkiverse/temporal/graal/grpchandling"
+# Directory structure for destination files in our extension
+# Where the modified files will be placed
+DEST_NETTY_RUNTIME="extension/runtime/src/main/java/io/quarkiverse/temporal/graal/netty"
+DEST_GRPC_RUNTIME="extension/runtime/src/main/java/io/quarkiverse/temporal/graal/grpc"
 
+# Deployment source directories in Quarkus
 SRC_NETTY_DEPLOYMENT="quarkus/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment"
 SRC_GRPC_DEPLOYMENT="quarkus/extensions/grpc-common/deployment/src/main/java/io/quarkus/grpc/common/deployment"
 
-DEST_NETTY_DEPLOYMENT="extension/deployment/src/main/java/io/quarkiverse/temporal/deployment/graal/nettyhandling"
-DEST_GRPC_DEPLOYMENT="extension/deployment/src/main/java/io/quarkiverse/temporal/deployment/graal/grpchandling"
+# Deployment destination directories in our extension
+DEST_NETTY_DEPLOYMENT="extension/deployment/src/main/java/io/quarkiverse/temporal/deployment/graal/netty"
+DEST_GRPC_DEPLOYMENT="extension/deployment/src/main/java/io/quarkiverse/temporal/deployment/graal/grpc"
 
-# When bumping temporal we need to check if netty was updated and match with Quarkus version
-QUARKUS_VERSION=3.16.3
+# Quarkus version to use - this should match the version compatible with Temporal
+QUARKUS_VERSION=3.19.3
```

Comment:
Don't update the quarkus version, as this will bring substitions for netty that are incompatible with grpc netty shaded 1.69.1 (which is the one being used in Quarkus 3.19.3).
Grpc Java 1.69.1 uses netty 4.1.110.Final, so the substitutions for Quarkus 3.19.3 target netty 4.1.118.Final, there we have substitition for `ResumptionController` that doesn't exist in the shaded netty. Right now the build didn't fail because it is still using the old `nettyhandling` and `grpchandling` packages, as the script wasn't run, and the old classes weren't removed.